qmd/sets.qmd

```diff
@@ -22,119 +22,351 @@ source("R/qmd-utils.R")
+  verify_input(
+    set_type_instruction,
+    glue::glue('Category instructions are missing for {category_set_ID}')
+  )
+  category_description <-
+    category_info[[parts_sheet_column_names$part_description_column_name]]
+  verify_input(set_type_description,
+               glue::glue('Category description is missing for {category_set_ID}'))
+  
+  category_display <- glue::glue(
+    '## {category_label} {{ #{category_set_ID} }}
+                              <a href=\"/parts.html#{category_set_ID}\">({category_set_ID})</a>. {category_description} {if (category_instruction %!=na% constants$dictionary_missing_value_replacement) category_instruction else "" }')
+  
+  # Create category sub categories
+  sub_categories_info_table <- categories_info_table[categories_info_table[[sets_sheet_column_names$part_set_ID_column_name]]== category_set_ID, ]
+  unique_sub_categories_parts <- unique(sub_categories_info_table[[parts_sheet_column_names$part_ID_column_name]])
+  sub_categories_list <- list()
+  for (sub_category_ID in unique_sub_categories_parts) {
+    sub_category_info <- parts_table[parts_table[[parts_sheet_column_names$part_ID_column_name]] == sub_category_ID, ]
+    # Verify that parts contains the requested part information
+    if (nrow(sub_category_info) < 1) {
+      warning(glue::glue('{sub_category_ID} has no valid part info and is skipped'))
+      next()
+    }
+    
+    # Declare display elements
+    sub_category_label <-
+      sub_category_info[[parts_sheet_column_names$part_label_column_name]]
+    sub_category_description <-
+      sub_category_info[[parts_sheet_column_names$part_description_column_name]]
+    sub_category_status <-
+      sub_category_info[[parts_sheet_column_names$part_status_column_name]]
+    sub_category_first_released <-
+      sub_category_info[[parts_sheet_column_names$part_first_release_column_name]]
+    sub_category_last_updated <-
+      sub_category_info[[parts_sheet_column_names$part_last_updated_column_name]]
+    
+    # Verify display elements
+    verify_input(sub_category_label,
+                 glue::glue('Part label is missing for {sub_category_ID}'))
+    verify_input(sub_category_description,
+                 glue::glue('Part description is missing for {sub_category_ID}'))
+    verify_input(sub_category_status,
+                 glue::glue('Part status is missing for {sub_category_ID}'))
+    verify_input(
+      sub_category_first_released,
+      glue::glue('Part first released is missing for {sub_category_ID}')
+    )
+    verify_input(sub_category_last_updated,
+                 glue::glue('Part last updates is missing for {sub_category_ID}'))
+    
+    sub_categories_list[[sub_category_ID]] <-
+      glue::glue(
+        '**{sub_category_label}** <a href=\"/parts.html#{sub_category_ID}\">({sub_category_ID})</a>. {sub_category_description} </br>Status: {sub_category_status}. First released: {sub_category_first_released}. Last updated: {sub_category_last_updated}.'
```

Comment:
This is the same template string as on lines https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L195 and https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L280. Probably should be moved into a function.